### DISCUSSION TITLE
useMutation where no variables are required

### DISCUSSION THREAD

**Discussion by danieljmann96** [TanStack/query#3974] [Q&A]
I've been using react-query for a while now and I'm loving it!
One (very) small issue I have with useMutation is the need to pass it an empty object even when the mutation function doesn't have any params, at least for TypeScript anyway. My use case is I have a booking app, and there is a function that just permanently deletes anything soft deleted longer than 2 weeks ago. This requires no parameters. This is currently what I'm doing:
```
clearOldBookings.mutate({})
```
The below would be preferable but gives a type error, as the "args" parameter is required in "mutate"
```
clearOldBookings.mutate()
```

> Arguments for the rest parameter 'args' were not provided.

What is the best practice for doing mutations without parameters?

**Comment by TkDodo** [comment]
works fine for me as variables are inferred to `void`

[typescript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgVwM4FMCyyYEMbAQB2cAvnAGZQQhwDkAAnkangMYDWA9FOrmzAC0AR2TooATzoAoaW2KtEcEDnzoycALwoM2PAWIAKQwEotAPjgAFaiGAYAdL1QQANgDd0hgKwmTslX0vEyA)

**Comment by robdodson** [comment]
@TkDodo If you want to use the onSuccess/onError callbacks, it seems like you have to pass undefined to mutate(). Do you know if there's a way to avoid that?

[typescript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgVwM4FMCyyYEMbAQB2cAvnAGZQQhwDkAAnkangMYDWA9FOrmzAC0AR2TooATzoAoaW2KtEcEDnzoycALwoM2PAWIAKQwEotAPjgAFaiGAYAdL1QQANgDd0hgKwmTslX0vJGIAZWQ2NnRUVAAuOFMLOHkWN3QHVwgAc0M6VAiomLo4E1J-QLVDZCIAE3QKYCJ0GoAaJTCC6LiEs01LFJdXdMycvM6ikrKgA)

**Reply by TkDodo** [reply]
no, I don't think so. you can call `mutate()` without passing anything, but if you need the second parameter (object with callbacks), you need to provide the first one as `undefined`.

**Reply by robdodson** [reply]
Makes sense. Thanks for the response, and for all the work you're doing building and blogging about this library!
